### PR TITLE
docs: polish the project organization doc

### PR DESCRIPTION
**Pull request by Startrekzky**
### ⚠️ &nbsp;&nbsp;Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [x] I have `npm run build` and `npm run serve` locally before submitting this PR
- [x] I have read through the [Contributing](https://devlake.apache.org/community/) Documentation

# Summary
1. Restructure the doc
2. Simplify the wording and remove redundant parts
3. Clarify the uses for team and project
4. Update the picture for use-case-2
5. Change the position of this doc in the directory
6. Standardize to second and third person
7. Improve the format
and more.

### Does this close any open issues?
The refinement of #637 

### Screenshots
image

image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

**Review by KucherenkoSerhiy — APPROVED** [review]
The guide looks much cleaner, concise, and to the point with these changes! Great job!

**Review by d4x1 — APPROVED** [review]
LGTM
